src/utils/throttle.js

```diff
@@ -1,43 +1,25 @@
-import https from 'node:https'
 import throttledQueue from 'throttled-queue'
 
-let httpsAgent
 let throttle
-let customFetch
-export const configure = options => {
-  customFetch = options.custom_fetch
-
+export const createThrottleQueue = options => {
   throttle = throttledQueue(
     options.throttleLimit,
     options.throttleInterval,
     options.throttleStrict
   )
-
-  httpsAgent = new https.Agent({
-    keepAlive: options.httpKeepAlive,
-    keepAliveMsecs: options.httpKeepAliveInterval
-  })
 }
 
 const wrapFetch = async (url, options) => {
-  let response
-  if (customFetch) {
-    response = await throttle(async () => customFetch(url, options))
-  } else {
-    response = await throttle(async () => globalThis.fetch(url, options))
-  }
-  const responseBody = await response.text()
 
-  response.text = () => Promise.resolve(responseBody)
+  const { custom_fetch } = options
+
+  const resolvedFetch = custom_fetch ?? globalThis.fetch
+
+  const response = await throttle(async () => resolvedFetch(url, options))
 
   return response
 }
 
-const keepAliveFetch = async (url, options) => {
-  // eslint-disable-next-line no-param-reassign
-  options.agent = () => httpsAgent
-  return wrapFetch(url, options)
-}
+const throttleFetch = async (url, options) => wrapFetch(url, options)
+export default throttleFetch
 
-export const throttleFetch = async (url, options) =>
-  keepAliveFetch(url, options)
```

Comment:
```suggestion
export const throttleFetch = fetch => async (url, options) => {
  const throttle = _throttleQueue ?? createThrottledQueue(options)
  return throttle(() => fetch(url, options))
}
```

Reply:
Thank you for the feedback and suggestions